toggl-open-source/toggldesktop
# Windows app auto update improvements

### 📒 Description
Windows app auto-update process improvements

### 🕶️ Types of changes
- **Bug fix** (non-breaking change which fixes an issue)
- **Breaking change** (fix or feature that would cause existing functionality to change)

### 🤯 List of changes
- [x] Switch to Onova library for performing auto-updates (new auto-update flow, see #3853)
- [x] New cmd line argument for the installer for lightweight auto-update (no registry overwrites or process force-quit)
- [x] Delete the main executable to prevent it from being launched while an update is running
- [x] Remove database file migration from v7.1 app
- [x] Refactor About view to MVVM

### 👫 Relationships
Closes #3853

### 🔎 Review hints

# Test scenarios

1. Clean install of the new installer
- Fully uninstall any installed app
- Install 7.5.75 app from the pre-built installer linked below
- Check that the app works correctly

2. Auto-update from 7.5.46 to new
- skip, can't test until we do a dev release

3. Auto-update from new to new
  3.1. About view
  - Install 7.5.74 version from the link below
  - Verify that the app can be auto-updated to 7.5.75 from About view.
  3.2. App restart
  - Install 7.5.74 version from the link below
  - Wait until the new version is downloaded
  - Verify that the app is auto-updated with a simple app restart.
  3.3. Not fully downloaded
  - Install 7.5.74 version from the link below
  - Quickly restart the app before the update is fully downloaded and make sure the app works properly

4. Manual install of the new installer on top of 7.5.46
  - Manually install pre-built 7.5.74 or 7.5.75 on top of installed 7.5.46 version.

5. Manual install of the new installer on top of a new app
  - Manually install 7.5.75 or 7.5.74 on top of installed 7.5.75 or 7.5.74 version

6. Debug mode → no auto-updates
  - Run the app in Debug mode. Make sure no updates are being downloaded (check in the About view or by network activity).

7. MS Store app → no auto-updates
  - Run the app in `StoreDebug` or `StoreRelease` configurations. Make sure no updates are being downloaded (check in the About view or by network activity).

Randomly test a couple of these scenarios on a 32-bit version.

Manually built installers, hardcoded to update to a custom 7.5.75 version by the second link:
32-bit:
https://github.com/skel35/toggldesktop/releases/download/v7.5.74/TogglDesktopInstaller-v7.5.74.exe
https://github.com/skel35/toggldesktop/releases/download/v7.5.75/TogglDesktopInstaller-v7.5.75.exe

64-bit:
https://github.com/skel35/toggldesktop/releases/download/v7.5.74/TogglDesktopInstaller-x64-v7.5.74.exe
https://github.com/skel35/toggldesktop/releases/download/v7.5.75/TogglDesktopInstaller-x64-v7.5.75.exe
